autometa/taxonomy/ncbi.py

```diff
@@ -92,332 +94,309 @@ def __str__(self):
-            if parent_taxid == ancestor_taxid:
-                return True
-            ancestor_taxid = self.parent(ancestor_taxid)
-        return False
-
-
-def main():
-    import argparse
-    import logging as logger
-    logger.basicConfig(
-        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
-        datefmt='%m/%d/%Y %I:%M:%S %p',
-        level=logger.DEBUG)
-    parser = argparse.ArgumentParser(
-        description='Autometa NCBI utilities class')
-    parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
-    parser.add_argument(
-        '--query-taxid',
-        help='query taxid to test NCBI class functionality',
-        default=1222,
-        type=int,
-    )
-    parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
-
-
-if __name__ == '__main__':
-    main()
+        lineage_a_taxids = {ancestor.get("taxid")
+                            for ancestor in self.lineage(taxid_A)}
+        lineage_b_taxids = {ancestor.get("taxid")
+                            for ancestor in self.lineage(taxid_B)}
+        common_ancestor = lineage_b_taxids.intersection(lineage_a_taxids)
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given `taxid` is a positive integer or not. In case the
+        taxid provided is in the form of a string the function will cast the string as an integer
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in NCBI taxonomy databases - nodes.dmp, names.dmp or merged.dmp
+
+        Returns
+        -------
+        int
+            `taxid` if the `taxid` is a positive integer
+
+        Raises
+        ------
+        ValueError
+            Provided `taxid` is not a positive integer
+        """
+        # This checks if an integer has been added as str, eg. "562"
+        if isinstance(taxid, str):
+            invalid_chars = set(string.punctuation + string.ascii_letters)
+            invalid_chars.discard('.')
+            if {char for char in taxid if char in invalid_chars} or taxid.count('.') > 1:
```

Comment:
I think using some set membership logic and the built-in `set` function makes this more readable. What do you think?

```suggestion
            if set(taxid).intersection(invalid_chars) or taxid.count('.') > 1:
```

Reply:
Oh, yeah, definitely. This is much more comprehensible